### ISSUE TITLE
Change project to include gitflow like setup

### ISSUE THREAD

**Issue by wcchristian** [tharakadesilva/SlackUserPresenceManager#13]
i.e. create a develop branch, make that the main branch, only merge to master on official tagged release.

then

when working on a feature/enhancement, your branch should be called feature/my-branch-description-{numberOfIssue}

bugs would be bug/....

**Comment by wcchristian** [comment]
This way we can also have hotfix branches, release branches (when one person is working on a release and others need the codebase for the next version etc.) Just suggesting this to follow industry standard.

**Comment by tharakadesilva** [comment]
I would love to do this but I am going to need your help on this. Any tips?

**Comment by wcchristian** [comment]
Yeah for sure. I'll post back here in a few hours with a small guide

**Comment by wcchristian** [comment]
Sorry for the delay, but here is the info that I know.

1. Create a develop branch

1. Make develop the main branch

1. When a user works on something they should make a branch as follows feature/#00-my-feature where 00 is the number of the issue that they are fixing.

1.  then they should merge back to develop

1. Once we are ready for release, merge develop into master and tag the release.

Basically the scope of this issue would be get the develop branch going and then write up a doc in the github wiki explaining this.

Types of branches are

- feature - Basic feature added

- bug - bugfix

- hotfix - A quick branch for a quick release, something is seriously wrong and needs to be fixed quickly

- release - something like release/1.1 would be created when 1.1 isn't quite released but some devs have started working on 1.2 which would be develop also note that release/1.1 would be merged back into develop and deleted once 1.1 is released.
